### PR TITLE
Feature/refactor client

### PR DESCRIPTION
**Pull request by ersoykadir**
- Client code improved with testing all endpoints.
- Related adjustments and bug fixes while testing client code handled.

### CODE REVIEW

File: app/client/main.py [hunk 1]
```diff
@@ -1,83 +1,234 @@
...
-    contexts = requests.get(url, headers=HEADERS).json()
+def request(type, url, headers, payload, params):
+    if type == 'GET':
+        res = requests.get(url, headers=headers, json=payload, params=params)
+    elif type == 'POST':
+        res = requests.post(url, headers=headers, json=payload, params=params)
+    elif type == 'PUT':
+        res = requests.put(url, headers=headers, json=payload, params=params)
+    elif type == 'DELETE':
+        res = requests.delete(url, headers=headers, json=payload, params=params)
```
Comment (mdegis) [inline app/client/main.py]: Those lines can be simplified by using `getattr`

https://www.geeksforgeeks.org/python-getattr-method/

File: app/client/main.py [hunk 2]
```diff
@@ -1,83 +1,234 @@
...
+    else:
+        raise Exception('Invalid request type')
+    # print(res.status_code, res.json())
+    if res.status_code == 400 or res.status_code == 401 or res.status_code == 404:
+        print(res.status_code, res.json()['detail'])
```
Comment (mdegis) [inline app/client/main.py]: print statements are usually messy and hard to keep track, I highly suggest to use logging module

File: app/server/crud/context.py [hunk 3]
```diff
@@ -7,10 +7,16 @@
 class Context():
```
Comment (mdegis) [inline app/server/crud/context.py]: this is python2.7 syntax, you do not need `()` in the header (after python3).

File: app/server/utils/auth.py [hunk 4]
```diff
@@ -40,11 +40,13 @@ def validate_token(token: Annotated[str, HTTPBearer] = Depends(oauth2_scheme)):
     payload = decode_access_token(token.credentials)
     email: str = payload.get("sub")
     exp = payload.get("exp")
+    if email == 'admin':
```
Comment (mdegis) [inline app/server/utils/auth.py]: codebase both includes single and double quotes (`'` and `"`). I think it is better to use the one type, linters usually suggest doubles (`"`).

File: app/client/main.py [hunk 5]
```diff
@@ -1,83 +1,234 @@
...
-# 1.2 Client gets an access token
-# 1.3. Client sends request to server with access token
-# 2. Client checks their contexts
-# 3. Client creates a context
-# 4. Client creates an item
-# 5. Client checks their items
-# 6. Client updates an item
-
-ACCESS_TOKEN = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJlZXJzb295NjFAZ21haWwuY29tIiwiZXhwIjoxNjkyNjI1NzUyfQ.h5In0Z87O__a5ITQPCWTot6_2ngkzRrzwEyd_o_WzkA'
+ACCESS_TOKEN = os.environ.get('ADMIN_ACCESS_TOKEN')
```
Comment (engd) [inline app/client/main.py]: It is usually a good idea to use the same variable name for the application and the container.

File: app/client/main.py (hunk 1 above)
Comment (ersoykadir) [reply app/client/main.py]: Done!

File: app/client/main.py (hunk 2 above)
Comment (ersoykadir) [reply app/client/main.py]: Will look into logging in python. Will be handled in further prs

File: app/server/crud/context.py (hunk 3 above)
Comment (ersoykadir) [reply app/server/crud/context.py]: Done!

File: app/server/utils/auth.py (hunk 4 above)
Comment (ersoykadir) [reply app/server/utils/auth.py]: Done!

File: app/client/main.py (hunk 5 above)
Comment (ersoykadir) [reply app/client/main.py]: Done!

**Pull request merged**